crates/wasm_interp/src/execute.rs

```diff
@@ -247,62 +279,200 @@ impl<'a> ExecutionState<'a> {
+                self.value_stack.push(Value::I32(value as i32));
+            }
+            I32LOAD16S => {
+                let addr = self.get_load_address(module) as usize;
+                let mut bytes = [0; 2];
+                bytes.copy_from_slice(&self.memory[addr..][..2]);
+                let value = i16::from_le_bytes(bytes);
+                self.value_stack.push(Value::I32(value as i32));
+            }
+            I32LOAD16U => {
+                let addr = self.get_load_address(module) as usize;
+                let mut bytes = [0; 2];
+                bytes.copy_from_slice(&self.memory[addr..][..2]);
+                let value = u16::from_le_bytes(bytes);
+                self.value_stack.push(Value::I32(value as i32));
+            }
+            I64LOAD8S => {
+                let addr = self.get_load_address(module) as usize;
+                let mut bytes = [0; 1];
+                bytes.copy_from_slice(&self.memory[addr..][..1]);
+                let value = i8::from_le_bytes(bytes);
+                self.value_stack.push(Value::I64(value as i64));
+            }
+            I64LOAD8U => {
+                let addr = self.get_load_address(module) as usize;
+                let value = self.memory[addr];
+                self.value_stack.push(Value::I64(value as i64));
+            }
+            I64LOAD16S => {
+                let addr = self.get_load_address(module) as usize;
+                let mut bytes = [0; 2];
+                bytes.copy_from_slice(&self.memory[addr..][..2]);
+                let value = i16::from_le_bytes(bytes);
+                self.value_stack.push(Value::I64(value as i64));
+            }
+            I64LOAD16U => {
+                let addr = self.get_load_address(module) as usize;
+                let mut bytes = [0; 2];
+                bytes.copy_from_slice(&self.memory[addr..][..2]);
+                let value = u16::from_le_bytes(bytes);
+                self.value_stack.push(Value::I64(value as i64));
+            }
+            I64LOAD32S => {
+                let addr = self.get_load_address(module) as usize;
+                let mut bytes = [0; 4];
+                bytes.copy_from_slice(&self.memory[addr..][..4]);
+                let value = i32::from_le_bytes(bytes);
+                self.value_stack.push(Value::I64(value as i64));
+            }
+            I64LOAD32U => {
+                let addr = self.get_load_address(module) as usize;
+                let mut bytes = [0; 4];
+                bytes.copy_from_slice(&self.memory[addr..][..4]);
+                let value = u32::from_le_bytes(bytes);
+                self.value_stack.push(Value::I64(value as i64));
+            }
+            I32STORE => {
+                let (addr, value) = self.get_store_addr_value(module);
+                let unwrapped = value.unwrap_i32();
+                let mut target = &mut self.memory[addr..][..4];
+                target.write_all(&unwrapped.to_le_bytes()).unwrap();
```

Comment:
did you consider `target.copy_from_slice(&unwrapped.to_le_bytes());`?

Reply:
Oh. No I didn't actually. I can give it a go.

Reply:
Thanks for the suggestion, that's definitely nicer. I was having a problem with `std::fmt::Write` vs `std::io::Write` and this gets rid of one of them.